compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -513,6 +514,19 @@ public void visit(BLangTypeDefinition typeDefinition) {
 
         if (typeDefinition.flagSet.contains(Flag.ENUM)) {
             ((BEnumSymbol) typeDefinition.symbol).addAnnotations(annotSymbols);
+            if (typeDefinition.symbol.kind == SymbolKind.ENUM) {
```

Comment:
Do we need this check even after the check after line number 515?

Reply:
I added this check because I hadn't much idea about the all possible scenarios. I'll remove it and see.